Tests/Common/Data/QuoteBarConsolidatorTests.cs

```diff
@@ -97,6 +102,86 @@ public void AggregatesNewQuoteBarProperly()
             Assert.AreEqual(bar3.LastBidSize, quoteBar.LastBidSize);
             Assert.AreEqual(bar4.LastAskSize, quoteBar.LastAskSize);
             Assert.AreEqual(bar1.Value, quoteBar.Value);
+            Assert.AreEqual(bar4.EndTime, quoteBar.EndTime);
+        }
+
+        [Test]
+        public void AggregatesNewTimeSpanQuoteBarProperly()
+        {
+            QuoteBar quoteBar = null;
+            var creator = new QuoteBarConsolidator(TimeSpan.FromMinutes(2));
+            creator.DataConsolidated += (sender, args) =>
+            {
+                quoteBar = args;
+            };
+
+            var time = DateTime.Today;
+            var period = TimeSpan.FromMinutes(1);
+            var bar1 = new QuoteBar
+            {
+                Time = time,
+                Symbol = Symbols.SPY,
+                Bid = new Bar(1, 2, 0.75m, 1.25m),
+                LastBidSize = 3,
+                Ask = null,
+                LastAskSize = 0,
+                Value = 1,
+                Period = period
+            };
+            creator.Update(bar1);
+            Assert.IsNull(quoteBar);
+
+            var bar2 = new QuoteBar
+            {
+                Time = time + TimeSpan.FromMinutes(1),
+                Symbol = Symbols.SPY,
+                Bid = new Bar(1.1m, 2.2m, 0.9m, 2.1m),
+                LastBidSize = 3,
+                Ask = new Bar(2.2m, 4.4m, 3.3m, 3.3m),
+                LastAskSize = 0,
+                Value = 1,
+                Period = period
+            };
+            creator.Update(bar2);
+
+            // pushing another bar to force the fire
+            var bar3 = new QuoteBar
```

Comment:
Should we have an `Assert.IsNull(quoteBar)` after `creator.Update(bar2)` ?

Reply:
Yes there absolutely should be - and it fails when I included it. I decided not to bring that up here because it is a separate issue that I really don't understand. But try it yourself.
I mentioned this with Jared and gave up, because since that test will fail, that should mean that bars actually get delayed when emitted, but in live I tested it and they pop out on time. Is live data ALWAYS tick or something? So it is still delayed but only by 1 tick?  https://quantconnect.slack.com/archives/C8KNL9LEQ/p1518574661000015

Reply:
@mchandschuh Ah sorry, I `Assert.IsNotNull(quoteBar)`. Which is what ^ is referring to.
Why Assert.IsNull(quoteBar)? Surely after 2 bars it should be emitted? I don't understand why it doesn't so I left it out.